pytorch_lightning/trainer/data_loading.py

```diff
@@ -11,483 +11,137 @@
-        if not isinstance(dataloader, (DataLoader, CycleIterator)):
-            return dataloader
-
-        cycle_iterator: Optional[CycleIterator] = None
-
-        if isinstance(dataloader, CycleIterator):
-            cycle_iterator = dataloader
-            dataloader = dataloader.loader
-
-        if (
-            _fault_tolerant_training()  # injects components to track the state
-            or self._requires_distributed_sampler(dataloader)  # sets the distributed sampler
-            or mode == RunningStage.PREDICTING  # to track indices for the predictions
-            or self._accelerator_connector.use_ipu  # IPUs use a custom `DataLoader`
-        ):
-            sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
-            dataloader = _update_dataloader(dataloader, sampler, mode=mode)
-
-        if cycle_iterator is not None:
-            cycle_iterator.loader = dataloader
-            return cycle_iterator
-
-        return dataloader
-
-    def _resolve_sampler(self, dataloader: DataLoader, shuffle: bool, mode: Optional[RunningStage] = None) -> Sampler:
-        if self._requires_distributed_sampler(dataloader):
-            if not isinstance(dataloader.sampler, (SequentialSampler, RandomSampler)):
-                raise MisconfigurationException(
-                    "You seem to have configured a sampler in your DataLoader. This will be replaced"
-                    " by `DistributedSampler` since `replace_sampler_ddp` is True and you are using"
-                    " distributed training. Either remove the sampler from your DataLoader or set"
-                    " `replace_sampler_ddp=False` if you want to use your custom sampler."
-                )
-            return self._get_distributed_sampler(
-                dataloader, shuffle, mode=mode, overfit_batches=self.overfit_batches, **self.distributed_sampler_kwargs
-            )
-
-        return dataloader.sampler
-
-    @staticmethod
-    def _get_distributed_sampler(
-        dataloader: DataLoader,
-        shuffle: bool,
-        overfit_batches: Union[int, float],
-        mode: Optional[RunningStage] = None,
-        **kwargs: Any,
-    ) -> DistributedSampler:
-        """This function is used to created the distributed sampler injected within the user DataLoader."""
-        kwargs["shuffle"] = shuffle and not overfit_batches
-        kwargs.setdefault("seed", int(os.getenv("PL_GLOBAL_SEED", 0)))
-        cls = UnrepeatedDistributedSampler if mode == RunningStage.PREDICTING else DistributedSampler
-        sampler = cls(dataloader.dataset, **kwargs)
-        return sampler
+        rank_zero_deprecation(
+            "`TrainerDataLoadingMixin.prepare_dataloader` was deprecated in v1.6 and will be removed in v1.8."
+        )
+        return self._data_connector._prepare_dataloader(dataloader, shuffle, mode)
 
     def reset_train_dataloader(self, model: Optional["pl.LightningModule"] = None) -> None:
```

Comment:
these should just be moved to the Trainer, right? 
`trainer.reset_x_dataloader()` will then call into the data connector. Otherwise users can't call the reset manually.

Reply:
Using grep.app I only found two cases of users using `reset_train/val/test/predict_dataloader`, so I thought it might be okay to deprecate and make private. But if we still want to users to have this functionality, I can move them to the Trainer

Reply:
Done